### PR TITLE
tests: use `native` terminal where necessary 

### PR DESCRIPTION
**Pull request by benpicco**
### Contribution description

For some reason those only are [detected](https://ci.riot-os.org/details/9b9947d6b7b14c7b829272316f527555) by CI when disabling Kconfig builds (#20211)

### Issues/PRs references

same as #20212

### CODE REVIEW

File: tests/periph/eeprom/Makefile [hunk 1]
```diff
@@ -4,4 +4,8 @@ FEATURES_REQUIRED += periph_eeprom
 
 USEMODULE += shell_cmds_default  # provides reboot command
 
+ifeq (native, $(BOARD))
```
Comment (OlegHahm) [inline tests/periph/eeprom/Makefile:7]: The problem seems to be rather the `--eeprom` parameter not being passed properly to pyterm.

I propose to use
```diff
diff --git a/boards/native/Makefile.include b/boards/native/Makefile.include
index 0e7016caed..d886feb136 100644
--- a/boards/native/Makefile.include
+++ b/boards/native/Makefile.include
@@ -113,8 +113,12 @@ EEPROM_FILE ?= $(BINDIR)/native.eeprom
 
 # set the eeprom file flags only when the periph_eeprom feature is used.
 ifneq (,$(filter periph_eeprom,$(FEATURES_USED)))
-  EEPROM_FILE_FLAGS = --eeprom $(EEPROM_FILE)
-  TERMFLAGS += $(EEPROM_FILE_FLAGS)
+  ifeq (native,$(RIOT_TERMINAL))
+    EEPROM_FILE_FLAGS = --eeprom $(EEPROM_FILE)
+  else
+    EEPROM_FILE_FLAGS = '\-\-eeprom $(EEPROM_FILE)'
+  endif
+  TERMFLAGS += --process-args $(EEPROM_FILE_FLAGS)
 endif
 
 VCAN_IFNUM ?= 0

**Comment by OlegHahm** [comment]
To me it would sound sane to run all native tests without pyterm:
```diff
diff --git a/tests/Makefile.tests_common b/tests/Makefile.tests_common
index 5e988d903f..d96751c3ad 100644
--- a/tests/Makefile.tests_common
+++ b/tests/Makefile.tests_common
@@ -17,3 +17,7 @@ RIOTBASE ?= $(CURDIR)/../..
 QUIET ?= 1
 # DEVELHELP enabled by default for all tests, set 0 to disable
 DEVELHELP ?= 1
+
+ifeq (native, $(BOARD))
+  RIOT_TERMINAL ?= native
+endif

**Comment by benpicco** [comment]
> To me it would sound sane to run all native tests without pyterm

I agree, but we have many 'interactive' tests or tests that are more like examples where having `pyterm` is nice.

How about only disabling `pyterm` when *executing* the `test` target - #20215

**Comment by miri64** [comment]
I think the release tests need this as well: https://github.com/RIOT-OS/RIOT/actions/runs/7361658214/job/20039229745. :-/

**Comment by benpicco** [comment]
Closing this in favor of https://github.com/RIOT-OS/RIOT/pull/20215

**Pull request closed without merging**